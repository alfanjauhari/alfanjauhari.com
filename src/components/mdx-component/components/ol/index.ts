export * from './ol';
export * from './ol.styles';
