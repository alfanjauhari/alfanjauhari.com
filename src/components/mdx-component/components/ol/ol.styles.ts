import { styled } from '@/theme';

export const StyledOl = styled('ol', {
  my: '$5',
  counterReset: 'custom-counter',
  '& li': {
    counterIncrement: 'custom-counter',
    my: '$2',
    '&:before': {
      content: 'counter(custom-counter) ". "',
      mr: '$2',
      color: '$gray4',
    },
  },
});
