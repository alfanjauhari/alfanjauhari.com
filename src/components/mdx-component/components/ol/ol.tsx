import { OlHTMLAttributes } from 'react';
import { StyledOl } from './ol.styles';

export type OlProps = OlHTMLAttributes<HTMLOListElement>;

export function Ol({ start, ...props }: OlProps) {
  return (
    <StyledOl
      css={{
        counterSet: `custom-counter ${start ? start - 1 : 0}`,
      }}
      {...props}
    />
  );
}
