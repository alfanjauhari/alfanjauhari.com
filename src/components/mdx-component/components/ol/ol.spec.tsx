import { render, screen } from '@testing-library/react';
import { Ol } from './ol';

describe('MDXComponent/Ol', () => {
  it('should render successfuly', () => {
    render(
      <Ol>
        <li>First Item</li>
        <li>Second Item</li>
      </Ol>,
    );

    expect(screen.getByRole('list')).toBeVisible();
    expect(screen.getAllByRole('listitem')).toHaveLength(2);
  });

  it('should match snapshot', () => {
    render(
      <Ol>
        <li>First Item</li>
        <li>Second Item</li>
      </Ol>,
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
