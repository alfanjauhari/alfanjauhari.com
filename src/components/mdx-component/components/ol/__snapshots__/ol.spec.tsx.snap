// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Ol should match snapshot 1`] = `
<ol
  class="c-vUMlb c-vUMlb-ifxbzET-css"
>
  <li>
    First Item
  </li>
  <li>
    Second Item
  </li>
</ol>
`;
